corehq/apps/cloudcare/middleware.py

```diff
@@ -0,0 +1,22 @@
+from django.contrib import messages
+from django.contrib.auth.models import User
+from django.contrib.auth import login
+
+
+class CloudcareMiddleware(object):
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        auth_as = request.GET.get('auth_as', None)
+        view_name = '.'.join((view_func.__module__, view_func.__name__))
+        if request.user and request.user.is_superuser and auth_as:
+            try:
+                request.user = User.objects.get(username=auth_as)
```

Comment:
tiny but it'd be nice to just have to pass in the short part of the id and have HQ generate the domain-specific username for you (not sure how hard it is to get access to the domain or if that's even possible in all views)